### DISCUSSION TITLE
[Feature] App push notifications for Memories on this day

### DISCUSSION THREAD

**Discussion by AGJSpiegelenberg** [immich-app/immich#3981] [Feature Request]
### The feature

It would be great if it is possible to set push notifications in the app that there are Memories on this day.
Now users are not always opening the app daily day but find the memories feature awesome. They don't want to miss any memories so a notification if there are memories would be great.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by daniele-athome** [comment]
Just a technical note: being self-hosted, Immich (possibly) can't connect to Google/Apple notification servers, so this would have to be implemented via a daily timer or something similar. But that's fine, I guess - we are talking about 1 connection per day.

**Reply by jrasm91** [reply]
We had previously thought about polling notifications from the server when the background task runs (background backup)

**Reply by Blu3wolf** [reply]
Push notifications can be done for self-hosted apps. Pushover is one such example service. Granted, the whole setup is not self-hosted, but the option would be nice.

**Reply by andriykrefer** [reply]
I think this would be the better approach, but I would not mind if the notification came through email or a telegram bot for instance

**Comment by Rubeast** [comment]
I would really love this feature!

If native notification support is not possible. Might be worth checking out Apprise (https://github.com/caronc/apprise).

**Comment by RanKKI** [comment]
Then people will receive notifications every day if they have a large amount of photos.

**Reply by bo0tzz** [reply]
It'd obviously be a setting that can be disabled just like the other notifications ;)

**Reply by RanKKI** [reply]
Yeah, I know, there must be a setting to disable it.

Based on my observation and personal thoughts, most people have existing photos when they switch to Immich. This results in 'receiving notifications every day.' 

**1. Leave it open**
If the user keeps the notification on and opens the app every day, then the notification becomes useless, since the user opens the app every day already; they don't need a notification to remind them to open it.

**2. Turn it off**
They don't use it.

---

The only use case I see is for people with a small amount of photos (but eventually your photo library will get bigger). 

Widgets (#6082) are much better at reminding users about their memories.

**Reply by andriykrefer** [reply]
I guess It would be 1 notification per day not one per media (so it would not be notification**S** every day)

**Reply by andriykrefer** [reply]
> **1. Leave it open** If the user keeps the notification on and opens the app every day, then the notification becomes useless, since the user opens the app every day already; they don't need a notification to remind them to open it.
> 
> **2. Turn it off** They don't use it.

I have a lot a media in gphotos and get notifications often. I always open then, because it is fast and give a little nostalgia. That is my personal take. A possible setting would be summarize once a week, but personally I would not use this way,

**Reply by FahadLive** [reply]
> I have a lot a media in gphotos and get notifications often. I always open then, because it is fast and give a little nostalgia. That is my personal take. A possible setting would be summarize once a week, but personally I would not use this way,

This is exactly what I do too. Having a reminder to go through old media is refreshing. Hoping it to be a feature of immich...

**Comment by pedrojreis** [comment]
This is just **HUGE** as far as WAF (Wife Approval Factor) goes, she does subscribe to Google One to also store her photos (on pair with Immich) because Google have this feature and she told me that's saddening that Immich doesn't have this.

**Comment by muhammadyusuf-kurbonov** [comment]
Or you can suggest getting a Firebase Token to send Push Notifications via Firebase (like Telegram does). Those who has self-hosted the Immich (I think) can get the API key and add to .env file

**Comment by ncmtl** [comment]
Would a widget do the trick ?

**Reply by andriykrefer** [reply]
I dont think so, you would still have to remember to check to widget, and personally I would not waste screen real state on this

**Comment by sjtechy** [comment]
OneDrive sends an email with 'Memories from this Day' and a few sample photos then links to the photo library filtered on all photos from the current date, potentially also a few days before and after if there aren't many.
It usually puts a smile on my face and would love this feature for Immich!

**Comment by dusnoki** [comment]
I vote yes :)

**Comment by meichertom** [comment]
I'm really looking to replace Google Photos with Immich. The app/server is absolutely fantastic — hats off to the developers! There's just this one missing feature holding me back from making the full switch. Most importantly, my better half is very insistent on having this feature — and, as we all know, peace at home is priceless.

**Comment by muhammadyusuf-kurbonov** [comment]
https://unifiedpush.org/users/distributors/

Another option: self-hosted Push Providers (found in a new Matrix Client App - ElementX). I have no experience with them. I'm gonna try in the near future.

**Comment by Dylan10126** [comment]
What about the possibility of browser-based push notifications? I would think this would get around any issues with google/apple; and the added functionality of PC users also getting the ability to receive notifications as well.

**Comment by TheInfamousAlk** [comment]
+1 for this feature, the wife keeps using google photos because of this one feature.

**Comment by zolakt** [comment]
For Home Assistant users, I made a blueprint that sends this through Home Assistant. I would prefer a built-in support, but until then, this should work. https://github.com/zolakt/immich-memories-blueprint

**Comment by Jhon-Mosk** [comment]
I may not understand something correctly, but such notifications are possible.
![IMG_20251027_193416.jpg](https://github.com/user-attachments/assets/549164a2-3dd0-4fd1-b7be-2d40a88aac39)

**Reply by zolakt** [reply]
No one said it's not possible, but they haven't yet implemented the feature to send out daily notifications with memories

**Reply by james-maschal** [reply]
> I may not understand something correctly, but such notifications are possible.

Was there a specific service you used to get this to work? I was thinking of writing a python script to check Immich's API for created memories and let me know when I have one for the day. If you have a solid method I'd love to know!

**Reply by zolakt** [reply]
> Was there a specific service you used to get this to work? I was thinking of writing a python script to check Immich's API for created memories and let me know when I have one for the day. If you have a solid method I'd love to know!

You don't need to check anything manually. Immich API has a method to fetch memories. You just pass the current  date to it (probably not even needed for "Today"). Check the rest_command from my docs.

From that response you pick one memory and one asset inside that memory, to get the asset id. Then there is an API method to get the url of the thumbnail for that asset id. You just put the url of that api call in the push notification. Check my code, it's pretty simple, regardless of what language you use.

Their api is well documented. It's just 2 api calls. One from the script, to get the memories. And the other is basically just a url to the thumbnail which you send to the push notification.

**Comment by ismaildakrory** [comment]
I help this comes soon, it's one of the most lovely features in google photos.

**Reply by james-maschal** [reply]
I wrote a script to manually check the API for memories each day, and then send me a notification through Gotify (or any similar service like ntfy). I can upload it and link it if you're interested.

**Reply by daniele-athome** [reply]
Please do share :-)

**Reply by ismaildakrory** [reply]
Yes please that will be great, until we get official support, I think you managed the best approach.

**Reply by james-maschal** [reply]
I'm working on reworking the script to be more user friendly, and more appropriate for uploading here. I wasn't the most concerned with making it robust since I was the only one using it! I should have it done and uploaded by tonight or tomorrow.

**Reply by james-maschal** [reply]
I realize I didn't directly reply to this thread, but I commented below with the link to the script. Enjoy!

**Comment by james-maschal** [comment]
Its live! [Immich-Memories-Notification](https://github.com/james-maschal/Immich-Memories-Notification)
Let me know if you have any issues with it, I'm not a deep expert in this stuff, but I get by :)

**Reply by ismaildakrory** [reply]
Great, I'll give it a try today.

**Comment by ismaildakrory** [comment]
If anyone interested, I built "Memories" push notifications for Immich :

What it does:
  - Sends a push notification when you have photos from "this day" in previous years
  - Shows thumbnail preview + year info (e.g., "Memories from 2020, 2018")
  - Tap notification → opens immich app
  - Multi-user support for the whole family
  
  https://github.com/ismaildakrory/immich-memories-notify
  
  Built this mostly vibe coding! Been running smoothly on my setup. PRs and feedback welcome!

**Reply by ismaildakrory** [reply]
Sorry if this spams the discussion, But I rebuilt the script entirely with total new features, if anyone downloaded the previous one please redownload.
https://github.com/ismaildakrory/immich-memories-notify

**Comment by arne182** [comment]
The ntfy way is nice so that everyone can control the frequency.

**Comment by mimedo** [comment]
I highly support this feature! Any news?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by arne182** [comment]
I have dialed in https://github.com/ismaildakrory/immich-memories-notify to give me exactly the notification that I need. @mimedo try it out and give your feedback. Having such a feature already finished can allow for an easy pull request to get a similar functionality. @bo0tzz thanks for your work.

**Comment by ismaildakrory** [comment]
<img width="1172" height="1279" alt="image" src="https://github.com/user-attachments/assets/de8830f3-0831-4cb0-83e9-977da85251a3" />

I added some new features to the memory notify project, you are welcome to try the new version
https://github.com/ismaildakrory/immich-memories-notify

**Comment by Schoof-T** [comment]
Please make this an option of Immich, I'm trying to convert my family but they all need/want this feature.
